### PR TITLE
Property set fix

### PR DESCRIPTION
**Pull request by rajagopalans**
Add Update and Delete methods for Property sets into the client.
use the "Request" object for requests.
Clean up the testing code to use only public methods.

### CODE REVIEW

File: apstra/api_property_sets_test.go [hunk 1]
```diff
@@ -55,75 +55,60 @@ func TestCreateGetUpdateGetDeletePropertySet(t *testing.T) {
...
-			t.Fatal(err)
-		}
-		polished2, err := ps2.polish()
+		log.Printf("testing GetPropertySet() against %s %s (%s)", client.clientType, clientName, client.client.ApiVersion())
+		ps2, err := client.client.GetPropertySet(ctx, id1)
 		if err != nil {
 			t.Fatal(err)
 		}
-		if !equal(ps, polished2.Data) {
+		if !equal((*PropertySetData)(ps), ps2.Data) {
```
Comment (chrismarget-j) [inline apstra/api_property_sets_test.go:96]: unnecessary type assertion

File: apstra/api_property_sets_test.go [hunk 2]
```diff
@@ -55,75 +55,60 @@ func TestCreateGetUpdateGetDeletePropertySet(t *testing.T) {
...
 
-		log.Printf("testing getPropertySet() against %s %s (%s)", client.clientType, clientName, client.client.ApiVersion())
-		_, err = client.client.getPropertySet(ctx, id1)
+		log.Printf("Testing GetPropertySet() against %s %s (%s). This should fail", client.clientType, clientName, client.client.ApiVersion())
+		ps2, err = client.client.GetPropertySet(ctx, id1)
 		if err == nil {
-			t.Fatal("expected a 404 here, didn't get one")
+			t.Fatal("This Get Should have failed")
 		}
+		log.Printf("Error Received %s", err)
```
Comment (chrismarget-j) [inline apstra/api_property_sets_test.go:112]: We won't get here if there's an error.

File: apstra/client.go [hunk 3]
```diff
@@ -1520,6 +1503,23 @@ func (o *Client) GetPropertySetByLabel(ctx context.Context, in string) (*Propert
 	return ps.polish()
 }
 
+func (o *Client) CreatePropertySet(ctx context.Context, in *PropertySetData) (ObjectId, error) {
```
Comment (chrismarget-j) [inline apstra/client.go:1506]: Public methods should always have a Go Doc Comment.

File: apstra/client.go [hunk 4]
```diff
@@ -1520,6 +1503,23 @@ func (o *Client) GetPropertySetByLabel(ctx context.Context, in string) (*Propert
...
+
+// UpdatePropertySet returns *PropertySet representing the only property set with the given label,
+// or an error if multiple property sets share the label.
+func (o *Client) UpdatePropertySet(ctx context.Context, id ObjectId, in *PropertySetData) error {
+	return o.updatePropertySet(ctx, id, in)
+}
+
+// DeletePropertySet returns *PropertySet representing the only property set with the given label,
+// or an error if multiple property sets share the label.
+
```
Comment (chrismarget-j) [inline apstra/client.go:1518]: Blank line breaks the Go Doc Comment.

File: apstra/api_property_sets_test.go [hunk 5]
```diff
@@ -55,75 +55,60 @@ func TestCreateGetUpdateGetDeletePropertySet(t *testing.T) {
...
-		if err != nil {
-			t.Fatal(err)
-		}
-		polished1, err := ps1.polish()
-		if err != nil {
-			t.Fatal(err)
-		}
-		if !equal(ps, polished1.Data) {
-			t.Fatal("ps and ps1 are not equal")
+		log.Printf("Testing duplicate CreatePropertySet() against %s %s (%s)", client.clientType, clientName, client.client.ApiVersion())
```
Comment (chrismarget-j) [inline apstra/api_property_sets_test.go:73]: Turns out the duplicate name prohibition is a 4.1.2-only thing. We need to either detect the version and skip this test for earlier apstra, or delete the test altogether, same as we have done for `GetThingsByLabel()`.

I'm leaning toward axe the test.

**Review by chrismarget-j — CHANGES_REQUESTED** [review]

File: apstra/api_property_sets_test.go [hunk 6]
```diff
@@ -55,75 +55,60 @@ func TestCreateGetUpdateGetDeletePropertySet(t *testing.T) {
...
-		log.Printf("testing getPropertySetsByLabel() against %s %s (%s)", client.clientType, clientName, client.client.ApiVersion())
-		psSlice, err := client.client.GetPropertySetsByLabel(ctx, ps.Label)
+		log.Printf("testing GetPropertySetByLabel() against %s %s (%s)", client.clientType, clientName, client.client.ApiVersion())
+		ps1, err := client.client.GetPropertySetByLabel(ctx, ps.Label)
 		if err != nil {
-			t.Fatal(err)
-		}
-		if len(psSlice) != 1 {
-			t.Fatalf("expected 1 property set, got %d", len(psSlice))
+			log.Printf("Error Received %q", err)
```
[the remark this answers is not in the document]
Comment (rajagopalans) [reply apstra/api_property_sets_test.go:66]: This is some weird merge issue. I know I deleted this altogether

**Review by chrismarget-j — APPROVED** [review]

**Pull request merged**